include/rcpputils/find_and_replace.hpp

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCPPUTILS__FIND_AND_REPLACE_HPP_
+#define RCPPUTILS__FIND_AND_REPLACE_HPP_
+
+#include <string>
+
+namespace rcpputils
+{
+
+/// Find and replace all instances of a string with another string.
+/**
+ * \param[in] input The input string.
+ * \param[in] find The substring to replace.
+ * \param[in] replace The string to substitute for each occurance of `find`.
+ * \return A copy of the input string with all instances of the string `find` replaced with the
+ *   string `replace`.
+ */
+inline std::string
```

Comment:
I'd recommend making this a template function, which works with `std::basic_string` instead. That way it's more flexible, but also that way you can avoid ODR violation without using `inline`.

Reply:
Added template function in eb1ec92. I still have an inline function for `std::string` so that we can handle raw strings. Let me know if there's a better way to handle this.

Reply:
Raw strings? You mean like `const char *`?

Reply:
Yes. Basically, most usages in the tests won't compile.